### PR TITLE
Convex_hull_2: Add an adapter 

### LINKED ISSUE #3813: 2D Convex hull of indices
Using the code given here:
https://stackoverflow.com/questions/43682852/cgal-convex-hull-of-points-with-info/43732224?noredirect=1#comment97521387_43732224

provide an integrated solution to get the convex hull of anything using a point property map
(traits with property like in spatial sorting for example).

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

We add an adapter so that one can can compute the 2D convex hull of, for example indices, and access the 2D point via a property map.

### Todo
- [x] Add to the documentation

## Release Management

* Affected package(s): Convex_hull_2

* Issue(s) solved (if any): fix #3813

### CODE REVIEW

**Comment by afabri** [comment]
Is there a C++11 way to fill a vector with values 0..... n-1 ? In CGAL examples I saw the boost::Counting_iterator. I just found the boost::repeat_fun(). Or is a good old C style for loop just fine?

**Comment by lrineau** [comment]
[**@afabri**](https://github.com/afabri) commented on [Apr 3, 2019, 4:56 PM GMT+2](https://github.com/CGAL/cgal/pull/3833#issuecomment-479525842 "2019-04-03T14:56:52Z - Replied by Github Reply Comments"):
> Is there a C++11 way to fill a vector with values 0..... n-1 ? In CGAL examples I saw the boost::Counting_iterator. I just found the boost::repeat_fun(). Or is a good old C style for loop just fine?

In C++11: [`std::iota`]:
```c++
std::vector<int> vec(10);
std::iota(vec.begin(), vec.end(), 0);
```

[`std::iota`]: https://en.cppreference.com/w/cpp/algorithm/iota

**Comment by afabri** [comment]
> 
> 
> [**@afabri**](https://github.com/afabri) commented on [Apr 3, 2019, 4:56 PM GMT+2](https://github.com/CGAL/cgal/pull/3833#issuecomment-479525842):
> 
> > Is there a C++11 way to fill a vector with values 0..... n-1 ? In CGAL examples I saw the boost::Counting_iterator. I just found the boost::repeat_fun(). Or is a good old C style for loop just fine?
> 
> In C++11: [`std::iota`](https://en.cppreference.com/w/cpp/algorithm/iota):
> 
> ```c++
> std::vector<int> vec(10);
> std::iota(vec.begin(), vec.end(), 0);
> ```

mega cool!!

**Comment by lrineau** [comment]
```

**Comment by lrineau** [comment]
```
[ 12%] Building CXX object CMakeFiles/ch_from_cin_to_cout.dir/ch_from_cin_to_cout.cpp.o
In file included from /home/travis/build/CGAL/cgal/Convex_hull_2/examples/Convex_hull_2/convex_hull_indices.cpp:3:
/usr/local/include/CGAL/Convex_hull_traits_adapter_2.h:108:84: error: expected
      '>'
  ...typename Base_traits::Less_signed_distance_to_line_2 _2*>(this)->operato...
                                                         ^
/usr/local/include/CGAL/Convex_hull_traits_adapter_2.h:108:25: note: to match
      this '<'
      return static_cast<const typename Base_traits::Less_signed_distanc...
                        ^
1 error generated.
```
https://travis-ci.org/CGAL/cgal/jobs/515231785#L2703

**Comment by maxGimeno** [comment]
Travis reports :
> Differences in Convex_hull_2: Property_map are new and not committed.

It means you added a dependency to the package Property Maps and did not edit the file in package_info.

**Comment by afabri** [comment]
@sloriot I just uploaded the [precompiled documentation](https://cgal.geometryfactory.com/~afabri/Convex_hull_2/Convex_hull_2/index.html#title4)

File: Convex_hull_2/doc/Convex_hull_2/Convex_hull_2.txt [hunk 1]
```diff
@@ -79,6 +79,21 @@ functions such as `ch_bykat()`.
 
 \cgalExample{Convex_hull_2/ch_from_cin_to_cout.cpp}
 
+
+\section Convex_hull_2ExampleIndex  Example using a Property Map
+
+In the following example we have as input a vector of points,
+and we retrieve the indices of of the points which are on the convex hull.
```
Comment (sloriot) [inline Convex_hull_2/doc/Convex_hull_2/Convex_hull_2.txt]:
```suggestion
and we retrieve the indices of the points which are on the convex hull.
```

File: Convex_hull_2/doc/Convex_hull_2/Convex_hull_2.txt [hunk 2]
```diff
@@ -79,6 +79,21 @@ functions such as `ch_bykat()`.
...
+The convex hull function takes as fourth argument a traits class
+that must be model of the concept `ConvexHullTraits_2`. It provides
+predicates such as orientation tests.
+The class `Convex_hull_traits_adapter_2` in combination with a
+`Pointer_property_map`, is a model. The indices `i` are then "points",
```
Comment (sloriot) [inline Convex_hull_2/doc/Convex_hull_2/Convex_hull_2.txt]:
```suggestion
`Pointer_property_map`, is such a model. The indices `i` are then "points",
```

File: Convex_hull_2/include/CGAL/Convex_hull_traits_adapter_2.h [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+#include <CGAL/disable_warnings.h>
+
+#include <boost/call_traits.hpp>
+
+#include <CGAL/property_map.h>
+
+
+namespace CGAL{
+  
+using ::get;
```
Comment (sloriot) [inline Convex_hull_2/include/CGAL/Convex_hull_traits_adapter_2.h]:
```suggestion
```

**Comment by afabri** [comment]
@sloriot Succesfully tested in [5.0-Ic-15](https://cgal.geometryfactory.com/CGAL/testsuite/results-5.0-Ic-15.shtml)

**Comment by maxGimeno** [comment]
I confirm

**Comment by afabri** [comment]
@sloriot we should probably do the same for the 3D convex hull, as pointed out on [stackoverflow](https://stackoverflow.com/questions/56486969/cgal-3d-convex-hull-with-points-with-info)

**Comment by sloriot** [comment]
I wonder if using `Surface_mesh<std::size_t>` with an appropriate vertex point map would not work directly. (maybe not due to `Kernel_traits` being used to get `Vector_3`).

**Comment by sloriot** [comment]
No we need the traits to be adapted too. I guess what we have in extreme vertices should almost be it.

**Pull request merged**